src/ecdsa_sign.rs

```diff
@@ -0,0 +1,52 @@
+use crypto_bigint::{Encoding, NonZero, U256};
+use rand::{rngs::StdRng, Rng, SeedableRng};
+use starknet::core::{
+    crypto::{EcdsaSignError, ExtendedSignature},
+    types::FieldElement,
+};
+use starknet_crypto::{rfc6979_generate_k, sign, SignError};
+
+pub fn ecdsa_sign(
```

Comment:
Please rename it non_determinist_ecdsa_sign so we clearly see the difference with the normal one